### PR TITLE
Feat/user profile

### PR DESCRIPTION
**Pull request by bhavyaj12**
### Added user profile functionality

- Modified backend to fetch user using username
- Added fetch user profile, edit profile, fetch user posts using - createSlice and asyncThunk
- Added user page with posts, followers, following
- Add user profile API services 
- Added modal and modal flag in reducers, edit profile form inside modal

**Please check app is RESPONSIVE**

### What not to check
- Follow unfollow

### Preview: 
profile-pr

### CODE REVIEW

File: src/redux-reducers/userProfileSlice.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+} from "services";
+
+export const fetchUserProfile = createAsyncThunk(
+  "auth/fetchUserProfile",
+  async ({ username }, { rejectWithValue }) => {
+    console.log("from fetch user profile", username);
+    try {
+      const { data } = await fetchUserProfileService(username);
+      const { user } = data;
+      console.log("fetchUserProfile", user);
```
Comment (manojasarsa) [inline src/redux-reducers/userProfileSlice.js:15]: You can remove this console.log if no longer required

File: src/redux-reducers/userProfileSlice.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+);
+
+export const fetchUserPosts = createAsyncThunk(
+  "posts/fetchUserPosts",
+  async ({ username }, { rejectWithValue }) => {
+    console.log(username);
+    try {
+      const { data } = await fetchUserPostsService(username);
+      const { posts } = data;
+      console.log("From fetchUserPosts thunk", data);
```
Comment (manojasarsa) [inline src/redux-reducers/userProfileSlice.js:30]: This too.

**Review by Amandeep-sinha — COMMENTED** [review]
Looks Great to go ahead Bhavya.....however you can remove the mockman tab from the UI.

**Review by manojasarsa — APPROVED** [review]
The user profile, edit profile, and user posts functionality is working as expected.
The app is responsive.
One suggestion is to hide the Side NavBar in mobile view, as you are already toggling it through a button. Or you can provide buttons at the bottom to navigate.

image

**Pull request merged**